### PR TITLE
General improvements

### PR DESCRIPTION
**Pull request by gsanchezgavier**
- Bump depenencies
- Bump integration sdk to v3
- Bump go version
- Activate unit test for versions of go > 1.11
   - Arguments refactor needed

### CODE REVIEW

File: cmd/nri-ecs.go [hunk 1]
```diff
@@ -6,17 +6,35 @@ import (
...
+	"github.com/newrelic/infra-integrations-sdk/integration"
+	"github.com/newrelic/infra-integrations-sdk/log"
+
+	v3 "github.com/newrelic/nri-ecs/internal/metadata/v3"
+	"github.com/newrelic/nri-ecs/pkg/ecs"
+)
+
+var (
+	integrationName    = "com.newrelic.ecs"
+	integrationVersion = "1.3.1"
```
Comment (paologallinaharbur) [inline cmd/nri-ecs.go:19]: can we inject it with build flags?

Comment (gsanchezgavier) [reply cmd/nri-ecs.go:19]: yes! i didn't work on it yet to avoid to many changes in the same PR and incrementally working on this

Comment (gsanchezgavier) [reply cmd/nri-ecs.go:19]: btw this is not a new version , this is actually the current release version

**Review by paologallinaharbur — APPROVED** [review]
Only the version that could be injected at build time (but it was an issue already present)

everything else looks good!

File: cmd/nri-ecs.go (hunk 1 above)
Comment (paologallinaharbur) [reply cmd/nri-ecs.go:19]: your call, for that reason is approved already 😄

**Pull request merged**